protocol/l2-blob.md

```diff
@@ -0,0 +1,199 @@
+opcode to obtain the `versionedhash` of the BLOB.
+
+Note that the parameter `challengedOriginBlockNumber` in both the new challenge and resolve functions refers to the
+original L1 block number corresponding to the L2 block containing the BLOB being challenged. This change also requires
+that the `challenge_window` must be larger than the `sequence_window` to prevent cases where the challenge window has
+already passed by the time the sequencer submits the batch on L1
+
+[1]: https://github.com/ethereum-optimism/optimism/blob/develop/packages/contracts-bedrock/src/L1/DataAvailabilityChallenge.sol
+
+## Storage Requirement and BLOB Gas Cost
+
+According to the EIP-4844 specification, BLOBs must be kept for at least
+[MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS][2]
+epochs, which is around 18 days. The storage upper limit can be calculated using the formula:
+`BLOB_SIZE * MAX_BLOB_PER_BLOCK / BLOCK_TIME * MIN_EPOCHS_FOR_BLOB_SIDECARS_REQUESTS * SECONDS_PER_EPOCH`.
+Assuming `MAX_BLOB_PER_BLOCK = 6` and `BLOCK_TIME = 2`, the upper limit is approximately 618 GB.
+
+To serve data challenge purposes, we can reduce disk requirements even further. According to the Alt-DA
+[specification][3], data only needs to be available for `l2blobChallengeWindow + l2blobResolveWindow`.
+If `l2blobChallengeWindow` is 43200 seconds and `l2blobResolveWindow` is 3600 seconds, the disk requirement
+will be around 17 GB.
+
+These storage requirements are manageable for a commodity computer, and the cost of storing the data is minimal
+compared to L1 data upload costs. This cost can be covered by adjusting the L1 `blobBaseFeeScalar` value.
+
+[2]: https://github.com/ethereum/consensus-specs/blob/4de1d156c78b555421b72d6067c73b614ab55584/configs/mainnet.yaml#L148
+[3]: https://github.com/ethereum-optimism/specs/blob/main/specs/experimental/alt-da.md#data-availability-challenge-contract
+
+## Derivation
+
+Most of the derivation and reorg logic remains consistent with Alt-DA. However, instead of invalidating the entire
+batch of the challenged origin, only the transaction containing the BLOB that is challenged and fails to be resolved
+will be deleted. The basic workflow is as follows:
+
+1. When the op-node derives a batch from the `BatchQueue`, it collects all the BLOB commitments included in the batch.
+2. It tracks the challenge and resolution status for the collected BLOB hashes. If some BLOBs are challenged and fail
+to be resolved, the op-node should trigger a `ResetError` to initiate a reorg.
+3. If a reorg is triggered, the op-node will revert to an older block and rederive the chain. Upon encountering a
+block containing the expired BLOB hash, it will remove the corresponding transaction and pass the remaining block
+data to the EL.
+
+## Fault Proof
+
+The derivation pipeline integrates with fault proofs by adding additional hint types to the preimage oracle to query
```

Comment:
Why is it necessary? L2 state should have no dependency on BLOB, only BLOB hash.